src/elm/clinical.ts

```diff
@@ -217,34 +230,37 @@ export class Concept extends Expression {
   codes: any;
   display?: string;
 
-  constructor(json: any) {
+  constructor(json: ELM.Concept) {
     super(json);
     this.codes = json.code;
     this.display = json.display;
   }
 
   // Define a simple getter to allow type-checking of this class without instanceof
   // and in a way that survives minification (as opposed to checking constructor.name)
-  get isConcept() {
+  get isConcept(): boolean {
     return true;
   }
 
-  toCode(ctx: Context, code: any) {
+  toCode(ctx: Context, code: any): dt.Code {
     const system = ctx.getCodeSystem(code.system.name) || {};
     return new dt.Code(code.code, system.id, code.version, code.display);
   }
 
-  exec(ctx: Context) {
+  exec(ctx: Context): dt.Concept {
     const codes = this.codes.map((code: any) => this.toCode(ctx, code));
     return new dt.Concept(codes, this.display);
   }
 }
 
 export class CalculateAge extends Expression {
-  precision: string;
+  precision: ELM.DateTimePrecision;
 
-  constructor(json: any) {
+  constructor(json: ELM.CalculateAge) {
     super(json);
+    if (!json.precision) {
+      throw new Error('Precision is required');
+    }
```

Comment:
Hmm.. I wonder how to handle this case. It's marked as optional in the [type defs](https://github.com/csenn/cql-execution/blob/e7c3267eabba2a688617de3d735ba444b8f8478e/src/types/elm.d.ts#L2563), but we throw an error when it isn't provided. Does omitting this error break some assumptions we've made in this class? Based on the [logical specification](https://cql.hl7.org/04-logicalspecification.html#calculateage), looks like precision should be `0..1`. Ugh. This might snowball into types being sad in other functions, and I've now realized is yet again my fault. Let's discuss with @cmoesel what to do with this one.

And fine I'll come clean now. Yes, I just very recently realized the utility of the logical specification...

Reply:
OK.  So here's the thing:

* The logical spec (and XSD) do allow precision to be missing.
* Neither say how you should calculate the age when it's missing.  Which is quite unfortunate.
* BUT... I don't think there is any way in the CQL _language_ to create this type _without_ a precision, because the language exposes it via specific [unit-based functions](https://cql.hl7.org/09-b-cqlreference.html#calculateage) (`CalculateAgeInYears`, `CalculateAgeInMonths`, etc); there is no `CalculateAge` without a built-in precision unit.
* So in some ways, this conversation is moot, because, in theory, precision will never _actually_ be undefined.

What does that mean for us here? I guess I don't know -- but probably it means we should just do whatever is easiest. ;-)

Reply:
I think this goes back to the previous comment about possibly updating the XML spec to ensure that precision is always there if it should always be there... Interested to hear both of your thoughts on this one. At worse we could always just create some special ELM types locally that act as overrides, something like `interface CqlExecCalculateAge extends ELM.Expression` but that will get annoying in more than just a few case

Reply:
I think it makes sense to file an issue against CQL asking that this property either be made required, or that the behavior is defined for when a precision is not provided.